### PR TITLE
Configure Prometheus to select kubevirt prometheusrules

### PR DESCRIPTION
**Pull request by assafad**
**What this PR does / why we need it**:
Right now, when `KUBEVIRT_DEPLOY_PROMETHEUS=true` Prometheus is being deployed, but it doesn't collect Kubevirt prometheus rules, without applying something like [kubevirt/hack/cluster-deploy.sh#L66](https://github.com/kubevirt/kubevirt/blob/f24a4618b3d2fc4db7e5db38b35d2fccc6d11e96/hack/cluster-deploy.sh#L66). This PR suggests to have this solution on kubervirtci level, rather than every operator to implement it independently. 
**Which issue(s) this PR fixes**:
Fixes # [CNV-37526](https://issues.redhat.com/browse/CNV-37526)

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

File: cluster-up/cluster/k8s-provider-common.sh [hunk 1]
```diff
@@ -159,6 +159,13 @@ function wait_for_cdi_ready() {
     fi
 }
 
+# configure Prometheus to select kubevirt prometheusrules
+function configure_prometheus() {
```
Comment (ormergi) [inline cluster-up/cluster/k8s-provider-common.sh:163]: Does it take some time for the configuration to apply? 
If so please consider also implementing wait function.

Comment (assafad) [reply cluster-up/cluster/k8s-provider-common.sh:163]: From what I tested it was a one second operation, and anyway it doesn't block any future operations, so not sure it's needed.

Comment (ormergi) [reply cluster-up/cluster/k8s-provider-common.sh:163]: I meant to whatever `spec.ruleSelector` and `spec.ruleNamespaceSelector` change will trigger.

**Review by ormergi — APPROVED** [review]

**Review by brianmcarey — APPROVED** [review]
/approve

thanks @assafad !

**Pull request merged**